client/src/store/todoSlice.ts

```diff
@@ -0,0 +1,152 @@
+import {
+  createAsyncThunk,
+  createSlice,
+  type PayloadAction,
+} from "@reduxjs/toolkit";
+import {
+  createTodo,
+  deleteTodo,
+  fetchTodos,
+  updateTodo,
+  type PaginatedTodos,
+  type Todo,
+} from "../api/todos";
```

Comment:
```suggestion
import {
  createTodo,
  deleteTodo,
  fetchTodos,
  updateTodo,
} from "../api/todos" 
import type {
 PaginatedTodos,
 Todo,
} from "@/api/todos";
``` 
Импорты и методы бывает не удобно разлечать, можно так оставить, но моментами удобнее писать так